plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/weighers/CallableWeigher.kt

```diff
@@ -113,7 +148,11 @@ internal object CallableWeigher {
         // TODO: FE1.0 has logic that uses `null` for receiverIndex if the symbol matches every actual receiver in order to "prevent members
         //  of `Any` to show up on top". But that seems hacky and can cause collateral damage if the implicit receivers happen to implement
         //  some common interface. So that logic is left out here for now. We can add it back in future if needed.
-        if (bestMatchWeightKind == null) return shortcutWeight
+        if (bestMatchWeightKind == null) {
+            return if (abortOnReceiverMismatch)
```

Comment:
Should not we return null here in case of `abortOnReceiverMismatch == true`?

Reply:
No, the current implementation is the intended behavior. Let me rename this parameter to make this clear.